### DISCUSSION TITLE
[Improvement]: Data Object PHP Classes Names

### DISCUSSION THREAD

**Discussion by zangab** [pimcore/pimcore#10899] [Q&A / Support]
### Improvement description

I stumbled on some interesting stuff tbh.
I created different data object classes and wanted to change the "name" property of the class to be more descriptive to the user bc usually just the class ID is used. However if I change the `name` the class also gets renamed and I was rly like "wtf". 

Why is this linked to the classname? Why can't the PHP class be based on the ID, which can't be changed? What use-case does this "name" even have if I can not use it as some kind of label for the object classes?
Furthermore, are there any naming conventions concerning this?

<img width="600" alt="pimcore-data-object-classname" src="https://user-images.githubusercontent.com/9076592/143264605-f2c83ea3-d611-4604-85d8-7fc10065387e.png">

**Comment by BabovicT** [comment]
Maybe add label(title) and if it is set then use that for admin "name" and if not then still use Name field like it is now. And even set label to be equal to name like it is for php class name but make it editable so users can change that.
